lib/animina/narratives/resources/post.ex

```diff
@@ -0,0 +1,131 @@
+defmodule Animina.Narratives.Post do
+  @moduledoc """
+  This is the post resource.
+  """
+
+  alias Animina.Calculations
+  alias Animina.Changes
+
+  use Ash.Resource,
+    data_layer: AshPostgres.DataLayer,
+    notifiers: [Ash.Notifier.PubSub],
+    authorizers: Ash.Policy.Authorizer
+
+  attributes do
+    uuid_primary_key :id
+
+    attribute :content, :string do
+      constraints max_length: 8192
+      allow_nil? false
+    end
+
+    attribute :slug, :string, allow_nil?: false
+    attribute :title, :string, allow_nil?: false
+
+    create_timestamp :created_at
+    update_timestamp :updated_at
+  end
+
+  pub_sub do
+    module Animina
+    prefix "post"
+
+    broadcast_type :phoenix_broadcast
+
+    publish :create, ["created", [:user_id, nil]]
+    publish :update, ["updated", :id]
+    publish :destroy, ["deleted", [:id]]
+
+    publish_all :destroy, ["deleted", [:user_id, :id]]
+  end
+
+  relationships do
+    belongs_to :user, Animina.Accounts.BasicUser do
+      api Animina.Accounts
+      attribute_writable? true
+    end
+  end
+
+  validations do
+    validate present(:content)
+    validate present(:title)
+  end
+
+  actions do
+    defaults [:create, :update, :destroy]
+
+    read :read do
+      primary? true
+      pagination offset?: true, keyset?: true, required?: false
+    end
+
+    read :by_user_id do
+      pagination offset?: true, keyset?: true, required?: false
+
+      argument :user_id, :uuid do
+        allow_nil? false
+      end
+
+      filter expr(user_id == ^arg(:user_id))
+    end
+  end
+
+  code_interface do
```

Comment:
It should be impossible to create or update a slug. I haven't read through the whole code but I just want to be sure that a third party API call can not create it's own slug.